vep/src/main/resources/listVariants.py

```diff
@@ -24,21 +24,25 @@ def main():
     """
     print('Python version: %s' % platform.python_version())
 
-    # get the source and output directories
-    dataset_srcdir = f'{S3DIR}/variants/*/*/*'
-    ld_server_srcdir = f'{S3DIR}/ld_server/variants/*'
-    outdir = f'{S3DIR}/out/varianteffect/variants'
+    # just specify, potentially with a list only new datasets
+    dataset_srcdir = f'{S3DIR}/variants/ExSeq/*/*'
+    existing_variants = f'{S3DIR}/out/varianteffect/variants'
+    new_variants = f'{S3DIR}/out/new-variants'
+
 
-    # create a spark session
     spark = SparkSession.builder.appName('vep').getOrCreate()
 
-    # slurp all the variants across ALL datasets, but only locus information
-    # combine with variants in LD Server to make sure all LD Server variants go through VEP for burden binning
+    # reduce new datasets just to their variants
     dataset_df = get_df(spark, dataset_srcdir)
-    ld_server_df = get_df(spark, ld_server_srcdir)
+    df = dataset_df.dropDuplicates(['varId'])
+
+    # read in all the variants we already know about and put them in varId column
+    existing_variants_df = spark.read.csv(existing_variants, sep='\t', header=False)
+    existing_variants_df = existing_variants_df.select('_c5').withColumnRenamed('_c5', 'varId')
 
-    df = dataset_df.union(ld_server_df)\
-        .dropDuplicates(['varId'])
+    # find variants in new datasets that aren't already in existing datasets
+    df = broadcast(df).join(existing_variants_df, dataset_df.varId == existing_variants.varId,
+                                    "leftanti")
```

Comment:
Just as a curious note: is there a limit to the size that can be broadcast and if so what happens if you exceed it? Also did you test without broadcast and what happened / what was the difference?

Reply:
I did a tests at 1gb, 50gb, and 80gb, and they all worked and fairly fast too.  I don't think I tried without broadcast.

Reply:
I can do more structured testing or we can just "do it live" with the release.